table/block_based/block_based_table_reader.cc

```diff
@@ -1717,7 +1717,8 @@ void BlockBasedTable::RetrieveMultipleBlocks(
     FSReadRequest& req = read_reqs[req_idx];
     Status s = req.status;
     if (s.ok()) {
-      if (req.result.size() != req.len) {
+      if ((req.result.size() != req.len) ||
+          (req_offset + block_size(handle) > req.result.size())) {
```

Comment:
Just making sure, ```req_offset + block_size(handle) == req.result.size()``` should hold even for direct IO with padding right?

Reply:
with padding, should be `<=`